components/dashboard/src/settings/Preferences.tsx

```diff
@@ -6,28 +6,105 @@
 
 import { useContext, useState } from "react";
 import CheckBox from "../components/CheckBox";
-import CodeText from "../components/CodeText";
 import InfoBox from "../components/InfoBox";
 import { PageWithSubMenu } from "../components/PageWithSubMenu";
 import PillLabel from "../components/PillLabel";
 import SelectableCard from "../components/SelectableCard";
 import Tooltip from "../components/Tooltip";
-import golandLogo from '../images/golandLogo.svg';
-import ideaLogo from '../images/intellijIdeaLogo.svg';
-import vscode from '../images/vscode.svg';
-import vscodeInsiders from '../images/vscodeInsiders.svg';
+import ideLogos from '../images/ideLogos';
 import { getGitpodService } from "../service/service";
 import { ThemeContext } from "../theme-context";
 import { UserContext } from "../user-context";
 import settingsMenu from "./settings-menu";
 
 type Theme = 'light' | 'dark' | 'system';
 
+interface IdePreferences {
+    sectionTitle: string;
+    sectionSubtitle: string;
+
+    desktopIdeTitle: string;
+    desktopIdeSubtitle: string;
+    desktopIdeLabel?: string;
+    desktopIdeInfobox?: string;
+    desktopIdeFootnote?: string;
+
+    defaultIde: string;
+    defaultDesktopIde: string;
+
+    options: { [key: string]: IdeOption };
+}
+
+interface IdeOption {
+    title: string;
+    type: 'browser' | 'desktop';
+    logo: string;
+    tooltip?: string;
+    label?: string;
+}
+
+const idePreferences: IdePreferences = {
+    sectionTitle: "Default IDE",
+    sectionSubtitle: "Choose which IDE you want to use.",
+
+    desktopIdeTitle: "Open in Desktop IDE",
+    desktopIdeSubtitle: "Choose whether you would like to open your workspace in a desktop IDE instead.",
+    desktopIdeLabel: "Beta",
+    desktopIdeInfobox: 'While in beta, when you open a workspace using a JetBrains IDE you will need to use the following password: <span class="bg-gray-100 dark:bg-gray-700 px-1.5 py-0.5 rounded-md text-sm font-mono font-medium">gitpod</span>',
+    desktopIdeFootnote: '<p class="text-left w-full text-gray-500 dark:text-gray-400">The <strong>JetBrains desktop IDEs</strong> are currently in beta. <a href="https://github.com/gitpod-io/gitpod/issues/6576" target="gitpod-feedback-issue" rel="noopener" class="gp-link">Send feedback</a> · <a href="https://www.gitpod.io/docs/integrations/jetbrains" target="_blank" rel="noopener" class="gp-link">Documentation</a></p>',
```

Comment:
I like an idea of independent data, i.e. valued before, so we can add more IDEs easily.

But not sure about decoupling rendering. Maybe it is fine to way for meta deployment to remove beta flag or change some wording?

wdyt?

Reply:
@corneliusludmann Let's try without customisations to UI for now, and only keep IDE configurations. I think it would be more flexible and simpler to change UI element in the dashboard instead of writing html in config maps. I doubt JetBrains gets out of Beta soon, rather about Q2 next year.

Reply:
Most important that we can add new IDEs easily like PyCharm. It would be good to have an issue to create IDE proxy serving static files to provide images as a next step. Could you file it please? 🙏

Reply:
Thinking about it again, I find that at least the infobox has the same cycle as the IDE images and it would be super handy to be able to deliver the contents of this box at the same time as the images.

For example, we will change the implementation from Code With Me to an SSH connection soon. When we roll out this image, the message, that it needs the password is obsolete at the same time. Probably, we need another message like “you need to have installed the JetBrains Toolbox” etc. That would need a synced deployment with the meta team to get the proper message and the images deployed at the same time.

I'm totally fine with removing the title text etc (although I don't think it will do any harm to have the footnote text and the beta label in the configmap as well). But at least for the infobox, it would be really useful to have it in the configmap, in my opinion.

And if it's just that you're uncomfortable with HTML text being allowed, then we can put plaintext in there and drop the code styling of the password.

What do you think, @akosyakov? Would that be a compromise? If that does not convince you, no problem. Just let me know and I'll remove the UI texts as you suggested and keep only the IDE options.

Reply:
sounds good

Reply:
I have updated the pull request accordingly.